packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -505,88 +573,135 @@ export default class AllureReporter extends WDIOReporter {
+            const hookStep = hookExecutable.startStep(hook.title)
+            this._state.push(hookExecutable)
+            this._state.push(hookStep)
+        // if the hook is custom by default, it will be reported as test (not usual)
+        } else if (!(isAllHook || isEachHook) && !useCucumberStepReporter) {
+            const customHookTest = this._state.currentSuite.startTest(`hook:${hook.title}`)
+            this._state.push(customHookTest)
+        } else if (useCucumberStepReporter) {
+            this.onTestStart(hook)
         }
-
-        // add hook as test to suite
-        this.onTestStart(hook)
     }
 
     onHookEnd(hook: HookStats) {
-        const { disableMochaHooks, useCucumberStepReporter } = this._options
+        const { disableHooks, useCucumberStepReporter } = this._options
 
-        // ignore global hooks
+        // ignore global hooks or hooks when option is set in false
+        // any hook is skipped if there is not a suite created.
         if (!hook.parent || !this._state.currentSuite) {
             return
         }
 
-        const isMochaAllHook = isMochaAllHooks(hook.title)
-        const isMochaEachHook = isMochaEachHooks(hook.title)
-
-        if (disableMochaHooks && !isMochaAllHook && !this._state.currentAllureTestOrStep) {
-            return
-        }
-
-        // set beforeEach / afterEach hook (step) status
-        if (disableMochaHooks && isMochaEachHook) {
-            this._endTest(hook.error ? AllureStatus.FAILED : AllureStatus.PASSED, hook.error)
-            // mocha doesn't fire onTestEnd for failed beforeEach hook
-            // we have to force mark test as failed test
-            if (isMochaBeforeEachHook(hook.title) && hook.error) {
-                const fakeTestStat = Object.assign({}, hook, { title: hook.currentTest }) as TestStats
-                this._endTest(fakeTestStat.error ? AllureStatus.FAILED : AllureStatus.PASSED, fakeTestStat.error)
+        const isAllHook = isAllTypeHooks(hook.title) // if the hook is before* for mocha/jasmine
+        const isEachHook = isEachTypeHooks(hook.title) // if the hook is after* for mocha/jasmine
+
+        // if the hook is before/after from mocha/jasmine
+        if ((isAllHook || isEachHook) && !disableHooks) {
+            // getting the root step
+            const currentHookRootStep = this._state.pop()
+            // getting before/after hook element
+            const currentHookRoot = this._state.pop()
+            if (currentHookRootStep || currentHookRoot) {
+                if (currentHookRootStep instanceof AllureStep && currentHookRoot instanceof ExecutableItemWrapper) {
+                    // stage to finish for all hook.
+                    currentHookRoot.stage = currentHookRootStep.stage = Stage.FINISHED
+                    // set status values
+                    switch (hook.state) {
```

Comment:
This method is pretty basic, does not take into account steps result values, only cover step assertion or not, if not assertion the result is broken by default. Let me work on this method to increase the capabilities.

Reply:
@lararojasmr You don't need to do any extra work here. This is the cross-language and cross-framework behavior of allure. Status is "failed" if there is an assertion error, and "broken" in all other cases.

Reply:
But, fixtures do not work in the same way, because, if a hook return error undefined and non status, the result is not broken is passed. So the idea is validate when a step is related to hook or not so into a fixture the hook is an step as same as test. With that solution all fixtures get a Broken result.

Reply:
I've implemented an idea that i have, take a look and give me your comments.